dev-tools/omdb/src/bin/omdb/db/saga.rs

```diff
@@ -0,0 +1,182 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! `omdb db saga` subcommands
+
+use crate::Omdb;
+use crate::check_allow_destructive::DestructiveOperationToken;
+use crate::db::datetime_rfc3339_concise;
+use anyhow::Context;
+use async_bb8_diesel::AsyncRunQueryDsl;
+use clap::Args;
+use clap::Subcommand;
+use diesel::prelude::*;
+use nexus_db_model::Saga;
+use nexus_db_model::SagaNodeEvent;
+use nexus_db_queries::context::OpContext;
+use nexus_db_queries::db;
+use nexus_db_queries::db::DataStore;
+use nexus_db_queries::db::datastore::DataStoreConnection;
+use nexus_db_queries::db::datastore::SQL_BATCH_SIZE;
+use nexus_db_queries::db::pagination::Paginator;
+use nexus_db_queries::db::pagination::paginated;
+use tabled::Tabled;
+use uuid::Uuid;
+
+/// `omdb db saga` subcommand
+#[derive(Debug, Args, Clone)]
+pub struct SagaArgs {
+    #[command(subcommand)]
+    command: SagaCommands,
+}
+
+#[derive(Debug, Subcommand, Clone)]
+enum SagaCommands {
+    /// List running sagas
+    Running,
+
+    /// Inject an error into a saga's currently running node
+    Fault(SagaFaultArgs),
+}
+
+#[derive(Clone, Debug, Args)]
+struct SagaFaultArgs {
+    saga_id: Uuid,
+}
+
+impl SagaArgs {
+    pub async fn exec(
+        &self,
+        omdb: &Omdb,
+        opctx: &OpContext,
+        datastore: &DataStore,
+    ) -> Result<(), anyhow::Error> {
+        match &self.command {
+            SagaCommands::Running => cmd_sagas_running(opctx, datastore).await,
+
+            SagaCommands::Fault(args) => {
+                let token = omdb.check_allow_destructive()?;
+                cmd_sagas_fault(opctx, datastore, args, token).await
+            }
+        }
+    }
+}
+
+async fn cmd_sagas_running(
+    _opctx: &OpContext,
+    datastore: &DataStore,
+) -> Result<(), anyhow::Error> {
+    let conn = datastore.pool_connection_for_tests().await?;
+
+    let sagas =
+        get_all_sagas_in_state(&conn, steno::SagaCachedState::Running).await?;
+
+    #[derive(Tabled)]
+    struct SagaRow {
+        id: Uuid,
+        creator_id: Uuid,
```

Comment:
I'd suggest putting the current sec here instead of the creator.  That's almost always more relevant I think.